packages/x-tailwindcss/demo/src/components/xds-input-group.vue

```diff
@@ -0,0 +1,122 @@
+<template>
+  <XdsBaseShowcase
+    #default="{ cssClass, section, copyCssClassesToClipboard, removeClassPrefix }"
+    title="Input Group"
+    :sections="sections"
+  >
+    <div
+      :key="cssClass"
+      @click="copyCssClassesToClipboard"
+      @keydown="copyCssClassesToClipboard"
+      :class="cssClass"
+      title="Click me to copy CSS classes"
+      :disabled="section === 'Disabled'"
+    >
+      <CuratedIcon class="x-icon" />
+      <input
+        class="x-input"
+        :disabled="section === 'Disabled'"
+        :placeholder="
+          removeClassPrefix(cssClass, base).trim() ||
+          (section.includes('Line') && 'line') ||
+          'input-group'
+        "
+      />
+      <button
+        @click.stop="copyCssClassesToClipboard"
+        :disabled="section === 'Disabled'"
+        class="x-button x-input-group-button"
+      >
+        <CrossIcon class="x-icon" />
+      </button>
+      <button
+        @click.stop="copyCssClassesToClipboard"
+        :disabled="section === 'Disabled'"
+        class="x-button x-input-group-button x-input-group-button-rectangle"
```

Comment:
Not sure if it should be the rectangle button version in this case. This way the distance between the text "clear" is also bigger than the search icon's padding. We can also compare this with suggestionGroup component, although it is applied as a rectangle, the spacing generated with its hover state is pretty different.
Maybe @cris-crash can help with this doubt.

Reply:
That is the intention of the rectangle variant. The `input-group-button` is squared and without padding by default, thought to be used with an icon or a very small word. When we add a text like "clear" it needs padding and removing aspect-ratio constraint. That's what the rectangle variant does.